lib/utils/hostuser.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"bytes"
+	"os/exec"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+// man GROUPADD(8), exit codes section
+const GroupExistExit = 9
+
+// man USERADD(8), exit codes section
+const UserExistExit = 9
+const UserLoggedInExit = 8
+
+// GroupAdd creates a group on a host using `groupadd`
+func GroupAdd(groupname string) (exitCode int, err error) {
+	groupaddBin, err := exec.LookPath("groupadd")
+	if err != nil {
+		return -1, trace.Wrap(err, "cant find groupadd binary")
+	}
+	cmd := exec.Command(groupaddBin, groupname)
+	err = cmd.Run()
```

Comment:
For all these commands, it would be useful to capture their stdout/err and show it in the logs in case of errors, for troubleshooting.